src/main/java/screens/GraphScene.java

```diff
@@ -328,4 +330,60 @@ public INodeHandler getInfo() {
         return this.info;
     }
 
+    /**
+     * Determine the number of genome paths going through an edge.
+     * @param incNode the node edges will be drawn from.
+     * @param id the id of the node.
+     * @return An array with the number of paths going through each outgoing edge of the node.
+     */
+    private double[] determineEdgeWidth(Node incNode, int id) {
+        NodeGraph ng = NodeGraph.getCurrentInstance();
+        int[] outgoing = incNode.getOutgoingEdges();
+        double[] widths = new double[outgoing.length];
+        int maxInt = -1;
+        double maxX = -Double.MAX_VALUE;
+        for (int i = 0; i < outgoing.length; i++) {
+            DrawNode out = ng.getDrawNode(outgoing[i]);
+            if (out != null && out.getX() > maxX) {
+                maxX = out.getX();
+                maxInt = i;
+            }
+            widths[i] = getNumberOfDuplicates(id, outgoing[i]);
+        }
+        if (maxInt == -1) {
+            return widths;
+        }
+        int[] incoming = ng.getNode(outgoing[maxInt]).getIncomingEdges();
+        if (incoming.length < 2 || outgoing.length < 2) {
+            return widths;
+        }
+        maxX = ng.getDrawNode(id).getX();
+        for (int i = 0; i < incoming.length; i++) {
+            DrawNode dNode = ng.getDrawNode(incoming[i]);
+            if (dNode != null && Math.abs(maxX - dNode.getX()) >= 5) {
+                widths[maxInt] -= getNumberOfDuplicates(outgoing[maxInt], incoming[i]);
+            }
+        }
+        return widths;
+    }
+
+    /**
+     * Get the number of paths going through node with id that also go through node outId.
+     * @param id the id of the node an edge will be drawn from.
+     * @param outId the id of the node an edge with be drawn to.
+     * @return the number of paths going through both nodes.
+     */
+    private int getNumberOfDuplicates(int id, int outId) {
+        int count = 0;
+        int[][] paths = GraphInfo.getInstance().getPaths();
+        for (int j = 0; j < paths[id].length; j++) {
+            for (int k = 0; k < paths[outId].length; k++) {
+                if (paths[id][j] == paths[outId][k]) {
+                    count += 1;
```

Comment:
Maybe you could add a break to the inner for loop, because the when a genome is found, you can't find another one since they are unique.